airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java

```diff
@@ -137,39 +139,39 @@ private JsonNode statefulSplitSecrets(final UUID workspaceId, final JsonNode ful
    * @param oldConfig old full config
    * @param fullConfig new full config
    * @param spec connector specification
+   * @param validate should the spec be validated, tombstone entries should not be validated
    * @return partial config
    */
   private JsonNode statefulUpdateSecrets(final UUID workspaceId,
                                          final Optional<JsonNode> oldConfig,
                                          final JsonNode fullConfig,
-                                         final ConnectorSpecification spec)
+                                         final ConnectorSpecification spec,
+                                         final boolean validate)
       throws JsonValidationException {
-    validator.ensure(spec.getConnectionSpecification(), fullConfig);
+    if (validate) {
+      validator.ensure(spec.getConnectionSpecification(), fullConfig);
+    }
 
-    if (longLivedSecretPersistence.isPresent()) {
-      if (oldConfig.isPresent()) {
-        final var splitSecretConfig = SecretsHelpers.splitAndUpdateConfig(
-            workspaceId,
-            oldConfig.get(),
-            fullConfig,
-            spec,
-            longLivedSecretPersistence.get());
-
-        splitSecretConfig.getCoordinateToPayload().forEach(longLivedSecretPersistence.get()::write);
-        return splitSecretConfig.getPartialConfig();
-      } else {
-        final var splitSecretConfig = SecretsHelpers.splitConfig(
-            workspaceId,
-            fullConfig,
-            spec);
-
-        splitSecretConfig.getCoordinateToPayload().forEach(longLivedSecretPersistence.get()::write);
-
-        return splitSecretConfig.getPartialConfig();
-      }
-    } else {
+    if (longLivedSecretPersistence.isEmpty()) {
       return fullConfig;
     }
+
+    final SplitSecretConfig splitSecretConfig;
+    if (oldConfig.isPresent()) {
+      splitSecretConfig = SecretsHelpers.splitAndUpdateConfig(
```

Comment:
This feels like the conditional logic should be internal to the `SecretsHelper`.  Otherwise, it requires the caller to know when to use the different methods.  Because the `oldConfig` is an optional, the logic to determine how to split the config could be codified inside the method, simplifying the call from the client code (e.g. here).

Reply:
The only part that differs logically is the method from before are these three lines:
```
    if (validate) {
      validator.ensure(spec.getConnectionSpecification(), fullConfig);
    }
```
Everything else, including the section you called out, was existing logic that I _slightly_ cleaned up to remove some duplicate code.  I agree with what you're saying though and will make that change as part of this PR.

Reply:
@colesnodgrass No worries...it was hard to tell if it was pre-existing.  Fine to do that cleanup in a follow-on PR if that's easier.

Reply:
@jdpgrailsdev I think I will do it in a follow-up PR, I started pulling on this thread to clean this up and it started pulling in other unrelated code as well.